src/structurizr/model/element.py

```diff
@@ -98,10 +99,22 @@ def get_afferent_relationships(self) -> Iterator[Relationship]:
             r for r in self.get_model().get_relationships() if self is r.destination
         )
 
-    def add_relationship(self, relationship: Optional[Relationship] = None, **kwargs):
+    def add_relationship(
+        self, relationship: Optional[Relationship] = None, **kwargs
+    ) -> Relationship:
         if relationship is None:
             relationship = Relationship(**kwargs)
+        elif relationship in self.relationships:
+            return relationship  # Nothing more to do
```

Comment:
The comment is obvious but a docstring for the method would be nice.

```suggestion
            return relationship
```